File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int target = 1;

        for(int i = 0;i<nums.size();i++){
            if(nums[i] > 0 && nums[i]==target){
                target++;
            }else if(nums[i]>target){
                return target;
            }
        }
        return target;
    }
};

File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        Arrays.sort(nums);
        int ans = 1;
        for(int i=0; i<nums.length; i++) {
            if(nums[i] > 0 && ans == nums[i]) ans++;
            else if(nums[i] > ans) return ans;
        }
        return ans;
    }
}